pymeasure/instruments/rohdeschwarz/sfm.py

```diff
@@ -0,0 +1,1501 @@
+        "SOUR:FREQ:CHAN:SPEC ?",
+        "SOUR:FREQ:CHAN:SPEC %d",
+        """A int property controlling the current selected special channel number
+        valid selections are based on the country settings.
+        """,
+        )
+
+    #TODO REL --> REL[1..6]
+    def channel_up_relative(self):
+        """
+        Increases the outpout frequency to the next higher channel/special channel
+        based on the current country settings
+        """
+        Instrument.write(self,"SOUR:CHAN:REL UP")
+    #TODO REL --> REL[1..6]
+    def channel_down_relative(self):
+        """
+        Decreases the outpout frequency to the next low channel/special channel
+        based on the current country settings
+        """
+        Instrument.write(self,"SOUR:CHAN:REL DOWN")
+
+    channel_sweep_start = Instrument.control(
+        "SOUR:FREQ:CHAN:STAR?",
+        "SOUR:FREQ:CHAN:STAR %g",
+        """A float property controlling the start frequency for channel sweep in Hz
+
+        * Minimum 5 MHz
+        * Maximum 1 GHz
+        """,
+        validator = strict_range,
+        values = [5E6, 1E9]
+        )
+
+    channel_sweep_stop =  Instrument.control(
+        "SOUR:FREQ:CHAN:STOP?",
+        "SOUR:FREQ:CHAN:STOP %g",
+        """A float property controlling the start frequency for channel sweep in Hz
+
+        * Minimum 5 MHz
+        * Maximum 1 GHz
+        """,
+        validator = strict_range,
+        values = [5E6, 1E9]
+        )
+
+    channel_sweep_step = Instrument.control(
+        "SOUR:FREQ:CHAN:STEP?",
+        "SOUR:FREQ:CHAN:STEP %g",
+        """A float property controlling the start frequency for  channel sweep in Hz
+
+        * Minimum 5 MHz
+        * Maximum 1 GHz
+        """,
+        validator = strict_range,
+        values = [5E6, 1E9]
+        )
+    #TODO CW --> CW[1..6]
+    cw_frequency = Instrument.control(
+        "SOUR:FREQ:CW?",
+        "SOUR:FREQ:CW %g",
+        """A float property controlling the CW-frequency in Hz
+
+        * Minimum 5 MHz
+        * Maximum 1 GHz
+        """,
+        validator = strict_range,
+        values = [5E6, 1E9]
+        )
+    #TODO FIXED --> FIXED[1..6]
+    frequency = Instrument.control(
+        "SOUR:FREQ:FIXED?",
+        "SOUR:FREQ:FIXED %g",
+        """A float property controlling the frequency in Hz
+
+        * Minimum 5 MHz
+        * Maximum 1 GHz
+        """,
+        validator = strict_range,
+        values = [5E6, 1E9]
+        )
+
+    frequency_mode = Instrument.control(
+        "SOUR:FREQ:MODE?",
+        "SOUR:FREQ:MODE %s",
+        """A string property controlling which the unit is used in
+
+        Possible selections are:
+
+        ======  =======
+        Value   Meaning
+        ======  =======
+        CW      Continous wave mode
+        FIXED   fixed frequency mode
+        CHSW    Channel sweep
+        RFSW    Frequency sweep
+        ======  =======
+
+        _Note_: selecting the sweep mode, will start the sweep imemdiately!
```

Comment:
Great hint to add for the users, thanks!

Reply:
No problem, came across this formatting for the docstring when I was looking into it fo rthe tables.